**Problem 2:**

In [1]:
from pymongo import MongoClient
import pandas as pd

In [ ]:
# Connect Azure MongoDB (Cosmos DB API)
uri = "mongodb+srv://Suprithamflix:Password2002@bigdata-mflix.global.mongocluster.cosmos.azure.com/?tls=true&authMechanism=SCRAM-SHA-256&retrywrites=false&maxIdleTimeMS=120000"
client = MongoClient(uri)
print("Connected to Azure MongoDB successfully.")

# Verify available databases
print("\nDatabases available in Azure MongoDB:")
for db_name in client.list_database_names():
    print(" -", db_name)

C:\Users\supri\AppData\Local\Temp\ipykernel_25552\2505571124.py:3: UserWarning: You appear to be connected to a CosmosDB cluster. For more information regarding feature compatibility and support please visit https://www.mongodb.com/supportability/cosmosdb
  client = MongoClient(uri)
C:\Users\supri\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.13_qbz5n2kfra8p0\LocalCache\local-packages\Python313\site-packages\pymongo\synchronous\mongo_client.py:1758: UserWarning: You appear to be connected to a CosmosDB cluster. For more information regarding feature compatibility and support please visit https://www.mongodb.com/supportability/cosmosdb
  self._resolve_srv()


Connected to Azure MongoDB successfully.

Databases available in Azure MongoDB:
 - sample_supplies
 - sample_analytics
 - sample_mflix
 - sample_weatherdata
 - sample_training
 - sample_restaurants
 - sample_airbnb
 - mflix
 - sample_guides
 - sample_geospatial


Here i am connecting to Azure MongoDB CosmoDB API(Vcore).
These are the databases available in Azure MongoDB 
 - sample_supplies
 - sample_analytics
 - sample_mflix
 - sample_weatherdata
 - sample_training
 - sample_restaurants
 - sample_airbnb
 - mflix
 - sample_guides
 - sample_geospatial

 From this i am using Sample_mflix database for my Dashboard.

In [3]:
# Select the 'sample_mflix' database
db = client["sample_mflix"]

# Check collections in the database
print("\n Collections in 'sample_mflix':")
collections = db.list_collection_names()
for col in collections:
    print(" -", col)

# Inspect record counts for each collection
print("\nCollection document counts:")
for col in collections:
    count = db[col].count_documents({})
    print(f" - {col}: {count} documents")

# View a sample document from each collection
print("\nSample documents from collections:")
for col in collections:
    sample_doc = db[col].find_one()
    print(f"\nCollection: {col}")
    print(sample_doc)


 Collections in 'sample_mflix':
 - theaters
 - comments
 - embedded_movies
 - movies
 - sessions
 - users

Collection document counts:
 - theaters: 1564 documents
 - comments: 41079 documents
 - embedded_movies: 3483 documents
 - movies: 21349 documents
 - sessions: 1 documents
 - users: 185 documents

Sample documents from collections:

Collection: theaters
{'_id': ObjectId('59a47286cfa9a3a73e51e72c'), 'theaterId': 1000, 'location': {'address': {'street1': '340 W Market', 'city': 'Bloomington', 'state': 'MN', 'zipcode': '55425'}, 'geo': {'type': 'Point', 'coordinates': [-93.24565, 44.85466]}}}

Collection: comments
{'_id': ObjectId('5a9427648b0beebeb69579e7'), 'name': 'Mercedes Tyler', 'email': 'mercedes_tyler@fakegmail.com', 'movie_id': ObjectId('573a1390f29313caabcd4323'), 'text': 'Eius veritatis vero facilis quaerat fuga temporibus. Praesentium expedita sequi repellat id. Corporis minima enim ex. Provident fugit nisi dignissimos nulla nam ipsum aliquam.', 'date': datetime.datetime

Here i am loading data form Sample_mflix database.

- It has collection : theaters(1564 documents), comments (41079 documents), embedded_movies (3483 documents), movies (21349 documents), sessions (1 document), users(185 documents).
- It also has information about the sample documnets from the collection.
